### PR TITLE
Add spacing attributes to comment author avatar

### PR DESCRIPTION
**Pull request by cbravobernal**
## Description
With this PR, we are adding spacing attributes (margin and padding) to the Comment Author Avatar.
We include specific PHP code in order to apply some styles to the wrapper and other styles to the image tag.
Please, feel free to propose a better solution if possible, as I could not find any core function that provides the attribute non serialized in a style HTML attribute string.

Maybe is a good idea to include a function to return scoped styles and classes with supports fields in block.json attributes. For example:
`get_scoped_style_and_classes(['spacing', 'border']);

I would be happy to work on that function if is a good approach.

## How has this been tested?

- Add Comment Author Avatar
- Tested adding padding, margin, and removing them.
- Check code both on editor and frontend.

## Screenshots 
Editor:
editor

Frontend:
frontend.

### CODE REVIEW

File: packages/block-library/src/comment-author-avatar/block.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+	"$schema": "https://json.schemastore.org/block.json",
```
Comment (gziolo) [inline packages/block-library/src/comment-author-avatar/block.json]: Schema URL has changed as of https://github.com/WordPress/gutenberg/pull/35900:

```suggestion
	"$schema": "https://schemas.wp.org/trunk/block.json",
```

It should be fixed also in https://github.com/WordPress/gutenberg/pull/36351.

File: packages/block-library/src/comment-author-avatar/index.php [hunk 2]
```diff
@@ -25,6 +25,23 @@ function render_block_core_comment_author_avatar( $attributes, $content, $block
 
 	// This is the only way to retreive style and classes on different instances.
 	$wrapper_attributes = WP_Block_Supports::get_instance()->apply_block_supports();
+
+	/**
+	 * We get the spacing attributes and transform the array provided into a string formatted for being applied as a style html tag.
+	 * Good candidate to be moved to a separate function in core.
+	 *
+	**/
```
Comment (gziolo) [inline packages/block-library/src/comment-author-avatar/index.php]:
```suggestion
	 * Good candidate to be moved to a separate function in core.
	 */
```

File: packages/block-library/src/comment-author-avatar/index.php [hunk 3]
```diff
@@ -25,6 +25,23 @@ function render_block_core_comment_author_avatar( $attributes, $content, $block
...
+	$spacing_attributes = isset($attributes['style']['spacing']) ? $attributes['style']['spacing'] : null;
+	if ( isset($spacing_attributes) && ! empty( $spacing_attributes ) ) {
```
Comment (gziolo) [inline packages/block-library/src/comment-author-avatar/index.php]: There are formatting issues detected by CI:

https://github.com/WordPress/gutenberg/runs/4143515671?check_suite_focus=true#step:6:173

`npm run format-php` should fix code formatting.

**Review by gziolo — COMMENTED** [review]
It looks like this PR is on track. @oandregal, is there a better way for spacing handling in PHP? `useSpacingProps` usage in the `edit` function is very convenient, it would be super nice to have something similar in PHP.

File: packages/block-library/src/comment-author-avatar/index.php (hunk 3 above)
Comment (cbravobernal) [reply packages/block-library/src/comment-author-avatar/index.php]: Couldn't do it before as I was having weird docker errors. Fixed!

File: packages/block-library/src/comment-author-avatar/block.json (hunk 1 above)
Comment (cbravobernal) [reply packages/block-library/src/comment-author-avatar/block.json]: Updated in this PR as well :-)

**Review by gziolo — APPROVED** [review]
I think I missed this PR 😅

File: packages/block-library/src/comment-author-avatar/index.php [hunk 4]
```diff
@@ -45,7 +61,10 @@ function render_block_core_comment_author_avatar( $attributes, $content, $block
 			'class'      => $classes,
 		)
 	);
-	return $avatar_string;
+	if ( isset( $spacing_attributes ) ) {
+		return sprintf( '<div style="%1s">%2s</div>', $spacing_string, $avatar_block );
```
Comment (gziolo) [inline packages/block-library/src/comment-author-avatar/index.php:65]: One thing that came up to my mind. Should we sanitize the computed string for the style?

Comment (cbravobernal) [reply packages/block-library/src/comment-author-avatar/index.php:65]: Yep, seems a good idea. Should I create a new PR, right?

Comment (gziolo) [reply packages/block-library/src/comment-author-avatar/index.php:65]: Yes, that's the best way to move forward. It's also why a proper API would be better as we would have a central place to address this type of concers 😄

**Comment by oandregal** [comment]
>  is there a better way for spacing handling in PHP? useSpacingProps usage in the edit function is very convenient

Yeah, we should extract utility functions from the "apply" functions in block supports. Aaron had a PR that also needed it for typography https://github.com/WordPress/gutenberg/pull/36104

File: packages/block-library/src/comment-author-avatar/index.php (hunk 4 above)
Comment (gziolo) [reply packages/block-library/src/comment-author-avatar/index.php:65]: Fixed in https://github.com/WordPress/gutenberg/pull/36988, thank you!

**Pull request merged**